### ISSUE TITLE
linter: no-unnecessary-type-conversion autofix removes load-bearing Number() when declared types misrepresent runtime values

### ISSUE THREAD

**Issue by afonsojramos** [oxc-project/oxc#25549]
## What happened

Running `oxlint --fix` with `typescript/no-unnecessary-type-conversion` (oxlint 1.78.0, oxlint-tsgolint 7.0.2001, type-aware mode) removed `Number(...)` wrappers around Drizzle ORM numeric/decimal column reads. Drizzle declares these as `number` in some configurations while returning **strings at runtime**, so the "unnecessary" conversions were load-bearing: after the fix, balance arithmetic silently became string concatenation (`"100" + "-100"` → `"100-100"`). This was only caught by database integration tests; most affected sites had no coverage.

## Why this is a fixer-safety issue rather than a false positive

The rule's analysis is arguably correct given the declared types — the type inaccuracy is Drizzle's. But there's a categorical difference from its sibling rule:

- `no-unnecessary-type-assertion` removals are always runtime-safe: casts are compile-only.
- `no-unnecessary-type-conversion` removals change **runtime behavior** whenever the declared type misrepresents the value, which the linter cannot verify.

Type-lying at ORM/serialization boundaries is common enough (Drizzle numerics, raw SQL rows, JSON parsed as typed) that an automatic `--fix` here is a data-corruption hazard.

## Suggested change

Demote this rule's autofix to a suggestion (or mark it as a dangerous fix applied only under `--fix-dangerously`), so plain `--fix` doesn't silently alter runtime behavior in codebases where declared types and runtime values disagree.

## Repro sketch

```ts
// column typed as number by the ORM, but the driver returns "42.50"
declare const balance: number; // runtime value is actually the string "42.50"

const zeroed = Number(balance) === 0; // rule flags Number() as unnecessary
// after --fix: `balance === 0` — now a string comparison, always false
const total = Number(balance) + Number(balance); // after --fix: string concatenation
```

Happy to provide a full runnable repro with Drizzle if useful.

**Comment by camc314** [comment]
Hi, I couldn't reproduce this - it's marked as a suggestion, so only fixed with `--fix-suggestions`.

**Comment by camc314** [comment]
Hi, closing this out as not planned due the lack of ability to reproduce.

Thanks!
